services/storage/in_memory_storage.go

```diff
@@ -358,12 +359,97 @@ func (storage *InMemoryStorage) resetCommitStorage(blockHeight primitives.BlockH
 	storage.commitStorage[blockHeight] = views
 	return views
 }
+
 func (storage *InMemoryStorage) resetViewChangeStorage(blockHeight primitives.BlockHeight) map[primitives.View]map[MemberIdStr]*interfaces.ViewChangeMessage {
 	views := make(map[primitives.View]map[MemberIdStr]*interfaces.ViewChangeMessage)
 	storage.viewChangeStorage[blockHeight] = views
 	return views
 }
 
+func (storage *InMemoryStorage) GetMessagesLogs(blockHeight primitives.BlockHeight, view primitives.View) []interfaces.MemberMessagesLog {
+	memberMessages := make(map[MemberIdStr][]*protocol.BlockRef)
+	// get preprepare
+	if msg, exists := storage.GetPreprepareFromView(blockHeight, view); exists {
+		memberIdStr := MemberIdStr(msg.Content().Sender().MemberId())
+		msgs, _ := memberMessages[memberIdStr]
```

Comment:
a little verbose - you don't need the variable `msgs` if you change the append statement to work on `memberMessages[memberIdStr]`